javaLoginCore/src/main/java/com/hamdan/forzenbook/java/login/core/viewmodel/JavaLoginState.java

```diff
@@ -0,0 +1,55 @@
+package com.hamdan.forzenbook.java.login.core.viewmodel;
+
+import com.hamdan.forzenbook.java.core.Entry;
+import com.hamdan.forzenbook.java.core.ErrorOutcomes;
+
+public class JavaLoginState {
+    final private Entry email;
+    final private Entry code;
+    final private Boolean showInfoDialog;
+    final private Boolean inputtingCode;
+    final private Boolean isLoading;
+    final private Boolean hasError;
+
+    public JavaLoginState() {
+        email = new Entry("", ErrorOutcomes.NONE);
+        code = new Entry("", ErrorOutcomes.NONE);
+        showInfoDialog = false;
+        inputtingCode = false;
+        isLoading = false;
+        hasError = false;
+    }
+
+    public JavaLoginState(Entry inEmail, Entry inCode, Boolean dialogState, Boolean codeState, Boolean loading, Boolean errorState) {
+        email = inEmail;
+        code = inCode;
+        showInfoDialog = dialogState;
+        inputtingCode = codeState;
+        isLoading = loading;
+        hasError = errorState;
+    }
+
+    public Entry getEmail() {
+        return email;
+    }
+
+    public Entry getCode() {
+        return code;
+    }
+
+    public Boolean getShowInfoDialog() {
```

Comment:
I don't think object Booleans are needed here, just like the other file

Reply:
updated